tools/cmake/patches/140-freebsd-build.fix.patch

```diff
@@ -0,0 +1,29 @@
+--- a/Utilities/cmliblzma/common/tuklib_integer.h
++++ b/Utilities/cmliblzma/common/tuklib_integer.h
+@@ -42,6 +42,8 @@
+ #define TUKLIB_INTEGER_H
+ 
+ #include "sysdefs.h"
++#include <sys/param.h>
++#include <sys/types.h>
+ 
+ #if defined(__GNUC__) && defined(__GNUC_MINOR__)
+ #   define TUKLIB_GNUC_REQ(major, minor) \
+@@ -126,7 +128,7 @@
+ 		(((uint16_t)(num) << 8) | ((uint16_t)(num) >> 8))
+ #endif
+ 
+-#ifndef bswap32
++#if !defined(bswap32) || defined(BSD)
+ #	define bswap32(num) \
+ 		( (((uint32_t)(num) << 24)                       ) \
+ 		| (((uint32_t)(num) <<  8) & UINT32_C(0x00FF0000)) \
+@@ -134,7 +136,7 @@
+ 		| (((uint32_t)(num) >> 24)                       ) )
+ #endif
+ 
+-#ifndef bswap64
++#if !defined(bswap64) || defined(BSD)
```

Comment:
Consider replacing `defined(BSD)` with `defined(__FreeBSD__)`  instead.

Reply:
Same suggestions apply for `scripts/mkhash.c`, however consider reviewing [this patch](https://bugs.openwrt.org/index.php?do=details&task_id=971&order=category&sort=desc) posted ~3 years ago.